// @ngx-loading-bar/http-client/esm2020/http.module.mjs
import { LoadingBarModule } from '@ngx-loading-bar/core';
import { NgModule } from '@angular/core';
import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
import { LoadingBarInterceptor } from './loading-bar.interceptor';
import * as i0 from "@angular/core";
export class LoadingBarHttpClientModule {
}
LoadingBarHttpClientModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: LoadingBarHttpClientModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
LoadingBarHttpClientModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: LoadingBarHttpClientModule, imports: [HttpClientModule, LoadingBarModule], exports: [HttpClientModule, LoadingBarModule] });
LoadingBarHttpClientModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: LoadingBarHttpClientModule, providers: [
        {
            provide: HTTP_INTERCEPTORS,
            useClass: LoadingBarInterceptor,
            multi: true,
        },
    ], imports: [[HttpClientModule, LoadingBarModule], HttpClientModule, LoadingBarModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.1.1", ngImport: i0, type: LoadingBarHttpClientModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [HttpClientModule, LoadingBarModule],
                    exports: [HttpClientModule, LoadingBarModule],
                    providers: [
                        {
                            provide: HTTP_INTERCEPTORS,
                            useClass: LoadingBarInterceptor,
                            multi: true,
                        },
                    ],
                }]
        }] });
